File: tl/src/main/java/com/github/badoualy/telegram/tl/api/requests/TLRequestMessagesDeleteHistory.java

package com.github.badoualy.telegram.tl.api.requests;

import com.github.badoualy.telegram.tl.TLContext;
import com.github.badoualy.telegram.tl.core.TLMethod;
import com.github.badoualy.telegram.tl.core.TLObject;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

import static com.github.badoualy.telegram.tl.StreamUtils.readInt;
import static com.github.badoualy.telegram.tl.StreamUtils.readTLObject;
import static com.github.badoualy.telegram.tl.StreamUtils.writeInt;
import static com.github.badoualy.telegram.tl.StreamUtils.writeTLObject;


public class TLRequestMessagesDeleteHistory extends TLMethod<com.github.badoualy.telegram.tl.api.messages.TLAffectedHistory> {

    public static final int CLASS_ID = 0xb7c13bd9;

    public int getClassId() {
        return CLASS_ID;
    }


    public TLRequestMessagesDeleteHistory(        com.github.badoualy.telegram.tl.api.TLAbsInputPeer _peer,         int _maxId) {
        this.peer = _peer;
        this.maxId = _maxId;

    }



    public com.github.badoualy.telegram.tl.api.messages.TLAffectedHistory deserializeResponse(InputStream stream, TLContext context) throws IOException {

        TLObject res = readTLObject(stream, context);
        if (res == null) {
            throw new IOException("Unable to parse response");
        }
        if (res instanceof com.github.badoualy.telegram.tl.api.messages.TLAffectedHistory) {
            return (com.github.badoualy.telegram.tl.api.messages.TLAffectedHistory)res;
        }
        else {
            throw new IOException("Incorrect response type. Expected com.github.badoualy.telegram.tl.api.messages.TLAffectedHistory, got: " + res.getClass().getCanonicalName());
        }

    }
        


    protected com.github.badoualy.telegram.tl.api.TLAbsInputPeer peer;

    protected int maxId;


    public com.github.badoualy.telegram.tl.api.TLAbsInputPeer getPeer() {
        return peer;
    }

    public void setPeer(com.github.badoualy.telegram.tl.api.TLAbsInputPeer value) {
        this.peer = value;
    }

    public int getMaxId() {
        return maxId;
    }

    public void setMaxId(int value) {
        this.maxId = value;
    }


    @Override
    public void serializeBody(OutputStream stream) throws IOException {

        writeTLObject(this.peer, stream);
        writeInt(this.maxId, stream);
    }


    @Override
    public void deserializeBody(InputStream stream, TLContext context) throws IOException {

        this.peer = (com.github.badoualy.telegram.tl.api.TLAbsInputPeer)readTLObject(stream, context);
        this.maxId = readInt(stream);
    }



    @Override
    public String toString() {
        return "messages.deleteHistory#b7c13bd9";
    }

}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/request/TLRequestMessagesDeleteHistory.kt
package com.github.badoualy.telegram.tl.api.request

import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_CONSTRUCTOR_ID
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_INT32
import com.github.badoualy.telegram.tl.api.TLAbsInputPeer
import com.github.badoualy.telegram.tl.api.TLInputPeerEmpty
import com.github.badoualy.telegram.tl.api.messages.TLAffectedHistory
import com.github.badoualy.telegram.tl.core.TLMethod
import com.github.badoualy.telegram.tl.serialization.TLDeserializer
import com.github.badoualy.telegram.tl.serialization.TLSerializer
import java.io.IOException

/**
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
class TLRequestMessagesDeleteHistory() : TLMethod<TLAffectedHistory>() {
    @Transient
    var justClear: Boolean = false

    var peer: TLAbsInputPeer = TLInputPeerEmpty()

    var maxId: Int = 0

    private val _constructor: String = "messages.deleteHistory#1c015b09"

    override val constructorId: Int = CONSTRUCTOR_ID

    constructor(
            justClear: Boolean,
            peer: TLAbsInputPeer,
            maxId: Int
    ) : this() {
        this.justClear = justClear
        this.peer = peer
        this.maxId = maxId
    }

    @Throws(IOException::class)
    override fun deserializeResponse_(tlDeserializer: TLDeserializer): TLAffectedHistory = tlDeserializer.readTLObject(TLAffectedHistory::class, TLAffectedHistory.CONSTRUCTOR_ID)

    protected override fun computeFlags() {
        _flags = 0
        updateFlags(justClear, 1)
    }

    @Throws(IOException::class)
    override fun serializeBody(tlSerializer: TLSerializer) = with (tlSerializer)  {
        computeFlags()

        writeInt(_flags)
        writeTLObject(peer)
        writeInt(maxId)
    }

    @Throws(IOException::class)
    override fun deserializeBody(tlDeserializer: TLDeserializer) = with (tlDeserializer)  {
        _flags = readInt()
        justClear = isMask(1)
        peer = readTLObject<TLAbsInputPeer>()
        maxId = readInt()
    }

    override fun computeSerializedSize(): Int {
        computeFlags()

        var size = SIZE_CONSTRUCTOR_ID
        size += SIZE_INT32
        size += peer.computeSerializedSize()
        size += SIZE_INT32
        return size
    }

    override fun toString() = _constructor

    override fun equals(other: Any?): Boolean {
        if (other !is TLRequestMessagesDeleteHistory) return false
        if (other === this) return true

        return _flags == other._flags
                && justClear == other.justClear
                && peer == other.peer
                && maxId == other.maxId
    }
    companion object  {
        const val CONSTRUCTOR_ID: Int = 0x1c015b09.toInt()
    }
}
